server/config.yaml

```diff
@@ -1 +1,5 @@
-port: 4000
+port: 5432
```

Comment:
this port was supposed to be for the API, you should create another field that's specific for the database.
In fact you could make another level of fields to show that it's a part of the database, like so:

```
port: 4000
database:
  port: 5432
  host: "localhost"
  ..
```

for yaml files indentations are 2 spaces.


then, instead of moving the `ServerConfig` to to database, you should create another config struct, called something like `DatabaseConfig` with the same fields, and the `ServerConfig` struct in the main.go file should look like:

```go
type ServerConfig struct {
  Port int `yaml:"port"`
  DBConfig datastore.DatabaseConfig `yaml:"database"`
}
```

does that make sense? That might be why the login doesn't work anymore. Lemme know if you have any questions.